src/Microsoft.OData.Edm/Schema/EdmTargetPath.cs

```diff
@@ -0,0 +1,111 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmTargetPath.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <inheritdoc/>
+    public class EdmTargetPath : IEdmTargetPath
+    {
+        private List<IEdmElement> segments;
+        private readonly string path;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmTargetPath"/> class.
+        /// </summary>
+        /// <param name="segments">Target path segments.</param>
+        public EdmTargetPath(IEnumerable<IEdmElement> segments)
+        {
+            EdmUtil.CheckArgumentNull(segments, nameof(segments));
+
+            this.segments = segments.ToList();
+
+            ValidateSegments();
+
+            this.path = this.GetPathString();

```

Comment:
Are there scenarios where we'd want to work with an `EdmTargetPath` without accessing the `path` property? If there are such scenarios, then maybe we should create the path string lazily when it's first access in the property getter.